src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotJVMCIRuntime.java

```diff
@@ -499,7 +500,20 @@ T get() {
         }
     }
 
-    @NativeImageReinitialize private HashMap<Long, WeakReference<ResolvedJavaType>> resolvedJavaTypes;
+
+    static class KlassWeakReference extends WeakReference<ResolvedJavaType> {
+
+        private final Long klassPointer;
```

Comment:
I assume this is `Long` instead of `long` to avoid boxing in `expungeStaleEntries`?

Reply:
There's autoboxing in a more few places and I wanted to avoid having multiple copies of the box live so I promoted it to Long on entry to fromMetaspace.